src/stores/user.ts

```diff
@@ -51,26 +51,51 @@ export const useUserStore = defineStore("userStore", {
       }
     },
     async login(credentials: LoginCredentials) {
-      if (this.isLoggedIn) await logout();
-      const response = await login(credentials);
-      const authHeader = response.headers.authorization;
-      const authToken = authHeader.slice("Bearer ".length);
-      setAuthHeader(authHeader);
-      this.setUserData({ authToken, user: response.data });
-      router.push({ name: "HomePage" });
+      try {
+        if (this.isLoggedIn) await logout();
+        const response = await login(credentials);
+        const authHeader = response.headers.authorization;
+        const authToken = authHeader.slice("Bearer ".length);
+        setAuthHeader(authHeader);
+        this.setUserData({ authToken, user: response.data });
+        router.push({ name: "HomePage" });
+      } catch (err) {
+        console.error(err);
+      }
     },
     async register(credentials: RegisterCredentials) {
-      if (this.isLoggedIn) await logout();
-      const response = await register(credentials);
-      const authHeader = response.headers.authorization;
-      const authToken = authHeader.slice("Bearer ".length);
-      setAuthHeader(authHeader);
-      this.setUserData({ authToken, user: response.data });
+      try {
+        if (this.isLoggedIn) await logout();
+        const response = await register(credentials);
+        const authHeader = response.headers.authorization;
+        const authToken = authHeader.slice("Bearer ".length);
+        setAuthHeader(authHeader);
+        this.setUserData({ authToken, user: response.data });
+        router.push({ name: "UserPage" });
+      } catch (err) {
+        console.error(err);
+      }
+    },
+    async getUser() {
+      try {
+        if (this.isLoggedIn) await logout();
+        const response = await getUser();
+        const authHeader = response.headers.authorization;
+        const authToken = authHeader.slice("Bearer ".length);
+        setAuthHeader(authHeader);
+        this.setUserData({ authToken, user: response.data });
+      } catch (err) {
+        console.error(err);
+      }
```

Comment:
All three methods looks nearly the same. For sake of being DRY try to refactor that you'll extract shared logic into one method and then call that method with proper parameters.

Reply:
I hope (new commit) this is along the lines of what you had in mind?